app/test/vesting_test.go

```diff
@@ -0,0 +1,674 @@
+
+	// find and test a vesting account with endTime which is
+	// at least 10 seconds away time from now to give the tx enough time to complete
+	_, name, err := s.getAnUnusedDelayedVestingAccount(tmtime.Now().Unix() + 10)
+	assert.NoError(s.T(), err)
+
+	s.testTransferVestingAmount(name)
+}
+
+func (s *VestingModuleTestSuite) TestGenesisDelayedVestingAccountsDelegation() {
+	assert.NoError(s.T(), s.cctx.WaitForNextBlock())
+
+	// find and test a vesting account that has vesting (locked) balance
+	// 10 seconds is chosen to be on the safe side
+	_, name, err := s.getAnUnusedDelayedVestingAccount(tmtime.Now().Unix() + 10)
+	assert.NoError(s.T(), err)
+
+	s.testDelegatingVestingAmount(name)
+}
+
+func (s *VestingModuleTestSuite) TestGenesisDelayedVestingAccountsClaimDelegationRewards() {
+	assert.NoError(s.T(), s.cctx.WaitForNextBlock())
+
+	// find and test a vesting account with endTime which is
+	// 	at least 20 seconds away time from now
+	_, name, err := s.getAnUnusedDelayedVestingAccount(tmtime.Now().Unix() + 20)
+	assert.NoError(s.T(), err)
+
+	s.testDelegatingVestingAmount(name)
+	s.testClaimDelegationReward(name)
+}
+
+func (s *VestingModuleTestSuite) TestGenesisPeriodicVestingAccountsSpendableBalancePartiallyUnlocked() {
+	// Find a periodic vesting account that has some vested (unlocked) balance (its start time has already passed)
+	vAcc, name, err := s.getAnUnusedPeriodicVestingAccount(tmtime.Now().Unix() - 5)
+	assert.NoError(s.T(), err)
+	address := getAddress(name, s.cctx.Keyring).String()
+
+	// Since we want a partially unlocked balance we need to wait until
+	// the first period has passed if not already
+	for vAcc.GetVestedCoins(tmtime.Now()).IsZero() {
+		time.Sleep(time.Second)
+	}
+
+	block, err := s.cctx.LatestBlock()
+	assert.NoError(s.T(), err)
+	assert.NoError(s.T(), s.cctx.WaitForNextBlock())
+
+	balances, err := testfactory.GetAccountSpendableBalanceByBlock(s.cctx.GRPCClient, address, block)
+	assert.NoError(s.T(), err)
+
+	expectedSpendableBal := vAcc.GetVestedCoins(block.Time).AmountOf(app.BondDenom).Int64() + initBalanceForGasFee
+	assert.EqualValues(s.T(),
+		expectedSpendableBal,
+		balances.AmountOf(app.BondDenom).Int64(),
+		"spendable balance must match")
+}
+
+func (s *VestingModuleTestSuite) TestGenesisPeriodicVestingAccountsSpendableBalanceLocked() {
+	block, err := s.cctx.LatestBlock()
+	assert.NoError(s.T(), err)
+	assert.NoError(s.T(), s.cctx.WaitForNextBlock())
+
+	// Find a periodic vesting account that that is currently in a vesting (locked) state
+	// i.e. its start time yet to be reached.
+	vAcc, name, err := s.getAnUnusedPeriodicVestingAccount(block.Time.Unix() + 10)
+	assert.NoError(s.T(), err)
+	address := getAddress(name, s.cctx.Keyring).String()
+
+	balances, err := testfactory.GetAccountSpendableBalanceByBlock(s.cctx.GRPCClient, address, block)
+	assert.NoError(s.T(), err)
+
+	expectedSpendableBal := vAcc.GetVestedCoins(block.Time).AmountOf(app.BondDenom).Int64() + initBalanceForGasFee
```

Comment:
[question] if the balance is locked, is expectedSpendableBal == initBalanceForGasFee ? Should this test add that as an assertion?

Reply:
added